### ISSUE TITLE
[Flake test]pull-kubernetes-unit

### ISSUE THREAD

**Issue by zhangxiaoyu-zidif** [kubernetes/kubernetes#51303]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/sig testing
/priority failing-test
/kind bug

> Uncomment only one, leave it on its own line: 
>
> /kind bug
> /kind feature

**What happened**:
pull-kubernetes-unit always failed.

ref: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/51297/pull-kubernetes-unit/49166/

k8s.io/kubernetes/pkg/api TestRoundTripTypes 3.65s
e.g. log: 
E0824 23:52:24.743235   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.746361   12107 event.go:194] Unable to construct event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, InvolvedObject:v1.ObjectReference{Kind:"", Namespace:"", Name:"", UID:"", APIVersion:"", ResourceVersion:"", FieldPath:""}, Reason:"", Message:"", Source:v1.EventSource{Component:"", Host:""}, FirstTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, LastTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, Count:0, Type:""}': 'request construction error: '<nil>'' (will not retry!)
E0824 23:52:24.746681   12107 event.go:200] Server rejected event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, InvolvedObject:v1.ObjectReference{Kind:"", Namespace:"", Name:"", UID:"", APIVersion:"", ResourceVersion:"", FieldPath:""}, Reason:"", Message:"", Source:v1.EventSource{Component:"", Host:""}, FirstTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, LastTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, Count:0, Type:""}': '' (will not retry!)
E0824 23:52:24.746911   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.746964   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.747006   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.747050   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.747089   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.747123   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.747158   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.747193   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.747237   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.747274   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.747307   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.747343   12107 event.go:209] Unable to write event: 'unexpected object: <nil>' (may retry after sleeping)
E0824 23:52:24.747378   12107 event.go:144] Unable to write event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, InvolvedObject:v1.ObjectReference{Kind:"", Namespace:"", Name:"", UID:"", APIVersion:"", ResourceVersion:"", FieldPath:""}, Reason:"", Message:"", Source:v1.EventSource{Component:"", Host:""}, FirstTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, LastTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, Count:0, Type:""}' (retry limit exceeded!)
E0824 23:52:24.747796   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.747881   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.747941   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.748010   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.748073   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.748140   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.748188   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.748220   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.748255   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.748287   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.748318   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.748348   12107 event.go:209] Unable to write event: 'A weird error' (may retry after sleeping)
E0824 23:52:24.748378   12107 event.go:144] Unable to write event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, InvolvedObject:v1.ObjectReference{Kind:"", Namespace:"", Name:"", UID:"", APIVersion:"", ResourceVersion:"", FieldPath:""}, Reason:"", Message:"", Source:v1.EventSource{Component:"", Host:""}, FirstTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, LastTimestamp:v1.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, Count:0, Type:""}' (retry limit exceeded!)
E0824 23:52:24.752449   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.752565   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.752650   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.752711   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.984455   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.984611   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.984678   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.984740   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.984917   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.985006   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.985079   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.985145   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.985280   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.985372   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.985437   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.985505   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.985663   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.985738   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.985804   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.985876   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986018   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986087   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986139   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986205   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986362   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986438   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986505   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986573   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986746   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986816   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986884   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.986945   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)
E0824 23:52:24.987150   12107 event.go:209] Unable to write event: 'fake error' (may retry after sleeping)

k8s.io/kubernetes/pkg/kubectl/apps TestApps 0.00s

ref:
https://github.com/kubernetes/kubernetes/pull/51297
https://github.com/kubernetes/kubernetes/pull/51101

**Comment by liggitt** [comment]
```
PersistentVolumeList: diff: 
		object.Items[0].Spec.PersistentVolumeSource.HostPath.Type:
		  a: <nil>
		  b: (*api.HostPathType)(0xc42056d850)
```

**Comment by liggitt** [comment]
there is a defaulter that needs a corresponding default in the fuzzer: https://github.com/kubernetes/kubernetes/pull/46597/files#r135180954

**Comment by liggitt** [comment]
fixed in #51258
